Compilation Issue (-Werror)

First of all, I just want to say thank you for this great tool! I'm currently learning, and this project has been very helpful in my journey.

I ran into a compilation issue related to -Werror treating a warning as an error

```txt
rm -f -r build bin
gcc -std=gnu11 -Wall -Werror -Wno-unused-function -Wno-nonnull -I include src/about_preprocessor.c -c -o build/about_preprocessor.o
gcc -std=gnu11 -Wall -Werror -Wno-unused-function -Wno-nonnull -I include src/about_io.c -c -o build/about_io.o
gcc -std=gnu11 -Wall -Werror -Wno-unused-function -Wno-nonnull -I include src/about_arrays.c -c -o build/about_arrays.o
In file included from /usr/include/criterion/internal/assert.h:28,
                 from /usr/include/criterion/assert.h:1681,
                 from /usr/include/criterion/criterion.h:32,
                 from include/c_koans.h:1,
                 from src/about_arrays.c:1:
src/about_arrays.c: In function 'about_arrays_what_is_an_array_impl':
src/about_arrays.c:107:39: error: pointer 'yet_another_array' may be used after 'realloc' [-Werror=use-after-free]
  107 |         cr_assert_eq(yet_another_array[i], TODO,
      |                                       ^
src/about_arrays.c:96:10: note: call to 'realloc' here
   96 |     if (!realloc(yet_another_array, INIT_ARR_SIZE * sizeof(int))) {
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/about_arrays.c:103:30: error: pointer 'yet_another_array' may be used after 'realloc' [-Werror=use-after-free]
  103 |         if (yet_another_array[i] == INIT_ARR_SIZE + 1) {
      |                              ^
src/about_arrays.c:96:10: note: call to 'realloc' here
   96 |     if (!realloc(yet_another_array, INIT_ARR_SIZE * sizeof(int))) {
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/about_arrays.c:100:22: error: pointer 'yet_another_array' may be used after 'realloc' [-Werror=use-after-free]
  100 |     yet_another_array[INIT_ARR_SIZE] = 6;
      |                      ^
src/about_arrays.c:96:10: note: call to 'realloc' here
   96 |     if (!realloc(yet_another_array, INIT_ARR_SIZE * sizeof(int))) {
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make: *** [Makefile:36: build/about_arrays.o] Error 1
```


 Simply removing `-Werror` allows the project to compile successfully. My thought is that maybe this was intentional and that, at some point, we would fix it anyway. For now, this worked as a temporary fix for me.

I used docker to create a somewhat reproducible environment. Here are the Containerfile and the command line I used to run it.

```dockerfile
FROM ubuntu:24.04

RUN : \
    && apt-get update \
    && apt-get upgrade -y \
    && apt-get install --no-install-recommends -y \
        build-essential \
        findutils \
        make \
        libcriterion-dev \
    && rm -rf /var/lib/apt/lists/* \
    && :
```

```bash
# build
$ docker build -t c-koans-build - < Containerfile
```

--- a/.github/workflows/c_koans.yml
+++ b/.github/workflows/c_koans.yml
@@ -13,8 +13,6 @@ jobs:
     steps:
       - uses: actions/checkout@v2
 
-      - run: sudo add-apt-repository -y ppa:snaipewastaken/ppa
-
       - run: sudo apt-get update -qqq
 
       - run: sudo apt-get install -yqq criterion-dev

--- a/.github/workflows/c_koans.yml
+++ b/.github/workflows/c_koans.yml
@@ -8,7 +8,7 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-24.04
 
     steps:
       - uses: actions/checkout@v2

--- a/.github/workflows/c_koans.yml
+++ b/.github/workflows/c_koans.yml
@@ -15,6 +15,6 @@ jobs:
 
       - run: sudo apt-get update -qqq
 
-      - run: sudo apt-get install -yqq criterion-dev
+      - run: sudo apt-get install -yqq libcriterion-dev
 
       - run: make clean all

--- a/src/about_arrays.c
+++ b/src/about_arrays.c
@@ -92,8 +92,9 @@ Test(about_arrays, what_is_an_array)
 
     /*
         When we need to, we can make an array bigger to accommodate via realloc.
+        The address may change so we always reassign the pointer.
     */
-    if (!realloc(yet_another_array, INIT_ARR_SIZE * sizeof(int))) {
+    if (!(yet_another_array = realloc(yet_another_array, INIT_ARR_SIZE * sizeof(int)))) {
         exit(1);
     }
 

--- a/src/about_arrays.c
+++ b/src/about_arrays.c
@@ -31,7 +31,7 @@ Test(about_arrays, what_is_an_array)
     /*
      * An array variable's name is merely a label for the address of the first
      * element in the array.
-    */
+     */
     cr_assert_eq(*array, TODO,
         "Dereferencing this label's address gives us the "
         "value at that point");
@@ -94,7 +94,8 @@ Test(about_arrays, what_is_an_array)
         When we need to, we can make an array bigger to accommodate via realloc.
         The address may change so we always reassign the pointer.
     */
-    if (!(yet_another_array = realloc(yet_another_array, INIT_ARR_SIZE * sizeof(int)))) {
+    if (!(yet_another_array
+            = realloc(yet_another_array, INIT_ARR_SIZE * sizeof(int)))) {
         exit(1);
     }
 

